src/main/java/io/milvus/client/MilvusServiceClient.java

```diff
@@ -75,6 +77,10 @@ public MilvusServiceClient(@NonNull ConnectParam connectParam) {
             metadata.put(Metadata.Key.of("dbname", Metadata.ASCII_STRING_MARSHALLER), connectParam.getDatabaseName());
```

Comment:
why not use this metadata?

metadata.put(Metadata.Key.of("traceID", Metadata.ASCII_STRING_MARSHALLER), traceID);

Reply:
can you please help with an example how we can set traceId unique for every grpc call in metadata.

Reply:
i think user should just specify traceID but not know any idea about interceptor
MilvusClient can handle it

Reply:
@xiaofan-luan in that case can we expose a threadlocal variable in ConnectParam which can be set during client instantiation?

Reply:
I think use threadlocal make sense to me.
For each reqeust thread, they set threadlocal before request milvus.
the milvus rpc check the traceID threadLocal

Reply:
cool, I will raise a PR to set threadlocal variable in ConnectParam and further use it in HeaderAttachingClientInterceptor.